app/jobs/webhooks/facebook_events_job.rb

```diff
@@ -1,8 +1,28 @@
 class Webhooks::FacebookEventsJob < ApplicationJob
+  class LockAcquisitionError < StandardError; end
+
   queue_as :default
+  # https://edgeapi.rubyonrails.org/classes/ActiveJob/Exceptions/ClassMethods.html#method-i-retry_on
+  retry_on LockAcquisitionError, wait: 10.seconds, attempts: 5
 
   def perform(message)
     response = ::Integrations::Facebook::MessageParser.new(message)
-    ::Integrations::Facebook::MessageCreator.new(response).perform
+
+    lock_key = "#{response.sender_id}_#{response.recipient_id}"
```

Comment:
let's use the `lib/redis/redis_keys.rb`  and define a specific key pattern for the case to avoid key collision

Reply:
Noted

Reply:
Is the key appropriate? Will it be unique for that sender conversation

Reply:
this combination should be good. 👍 
even processing messages from `sender_id` alone sequentially should be okay.